projects/core/components/primitive-year-picker/primitive-year-picker.component.ts

```diff
@@ -169,7 +175,7 @@ export class TuiPrimitiveYearPickerComponent {
     itemIsInterval(item: number): boolean {
         const {value, hoveredItem} = this;
 
-        if (value === null || !this.isRange(value)) {
+        if (value === null || tuiIsArray(value) || !this.isRange(value)) {
```

Comment:
Unnecessary check, `!this.isRange(value)` already would return `false` for everything, except for the month range (value === null) is also unnecessary now that you've extended types.

Reply:
fixed